description of uc and ic and stuff (#8)
thesis20/thesis_paper: sections/csgcn/csgcn-chapter.tex

\subsection{Adjacency matrix and training}\label{subsec:csgcn_adj_adj_mat}
Compared to CSGCN-IS, context is now included in the graph as nodes and therefore also included in the adjacency matrix.
The graph is similar to the one in \Cref{fig:quadripartite-graph}, but context nodes are inserted between the sets of item and user nodes, and edges connect the nodes that have interacted in the context.
For the convolution, we want to preserve the context's initial embeddings from the first layer through to the last layer, since we found that this gave slightly better performance compared to also propagating information about neighbors to the context embeddings.
Therefore, the transposed matrices of $UC$ and $IC$ are not inserted into the adjacancy matrix.

> Additionally, we add the identity matrix $I$ to the last quadrant so that for each convolution, the context embeddings are only propagated with information about themselves.

Reviewer: ```suggestion
Additionally, we add the identity matrix $I$ to the last quadrant such that the context embeddings are only propagated with information about themselves in each convolution.
```

> Additionally, we add the identity matrix $I$ to the last quadrant such that the context embeddings are only propagated with information about themselves in each convolution.

Reviewer 2: ```suggestion
Instead, we add the identity matrix $I$ to the last quadrant such that the context embeddings are only propagated with information about themselves in each convolution.
```

\end{equation}
where $l$ is the current layer and $D$ is a $(|U| + |I| + |US| + |IS| + |C|) \times (|U| + |I| + |US| + |IS|+ |C|)$ degree matrix. 
We see that for this model, we only normalize on a row-level by multiplying by $D^{\frac{1}{2}}$ on the left side of the adjacency matrix.

> We found that for this version of the model, it gave slightly better performance compared to multiplying by $D^{\frac{1}{2}}$ on both sides of the adjacency matrix.

Reviewer: skal vi name droppe at det er den der random walk laplacian?

> We found that for this version of the model, it gave slightly better performance compared to multipluying with $D^{\frac{1}{2}}$ on both sides of the adjacency matrix.

Reviewer: ```suggestion
We found that for this version of the model, it gave slightly better performance compared to multiplying by $D^{\frac{1}{2}}$ on both sides of the adjacency matrix.
```

> The final embedding matrix after $l$ layers is obtained in the same way as CSGCN-IS where we take the mean of the nodes' representations at each layer.

Reviewer 2: men har vi faktisk kørt extensive forsøg med at erstatte med random walk og på begge sider?

Author: Men hvordan kan vi så forklare at vi gør det?

> The final embedding matrix after $l$ layers is obtained in the same way as CSGCN-IS where we take the mean of the nodes' representation at each layer.

Reviewer: ```suggestion
The final embedding matrix after $l$ layers is obtained in the same way as CSGCN-IS where we take the mean of the nodes' representations at each layer.
```
